compiler/ast.nim

```diff
@@ -196,6 +196,7 @@ type
     nkTypeOfExpr,         # type(1+2)
     nkObjectTy,           # object body
     nkTupleTy,            # tuple body
+    nkTupleClassTy,       # tuple type class
```

Comment:
You need to update lib/core/macros.nim too when you add a new AST kind. Note that this doesn't mean I'll accept this PR. I don't know yet.